### PR TITLE
Update recurring payment block styles

### LINKED ISSUE #12535: Styles within recurring payments block need some love
#### Steps to reproduce the issue

1. Add the new recurring payments block
2. Connect to Stripe
3. Go to add your first payment amount
4. There are some styles on this screen that can be cleaned up

#### What I expected

- The labels should be left aligned
- The padding/margin should be more consistent

#### What happened instead

- The labels are centered
- The margin is all over the shop

**Labels Before**

**Labels After**

**Margin Before**

**Margin After**

### PR DESCRIPTION
**Pull request by davemart-in**
This is a WIP PR. No need to review. I will update this description and the labels when I am ready for a review.

Fixes #12535

#### Changes proposed in this Pull Request:

The labels should be left aligned instead of centered. The padding/margin should be more consistent as well.

**BEFORE**

**AFTER**

#### Testing instructions:
1. Add the new recurring payments block
2. Connect to Stripe
3. Go to add your first payment amount
4. There are some styles on this screen that can be cleaned up

#### Proposed changelog entry for your changes:
* Updated recurring payment block styles

### CODE REVIEW

**Review by artpi — DISMISSED** [review]
Looks good!

**Review by jeherve — CHANGES_REQUESTED** [review]
This looks a lot better! I only have one remark below.

File: extensions/blocks/recurring-payments/edit.jsx [hunk 1]
```diff
@@ -306,8 +306,9 @@ class MembershipsButtonEdit extends Component {
 	renderDisclaimer = () => {
 		return (
 			<div className="membership-button__disclaimer">
+				{ __( 'Read more about ', 'jetpack' ) }
```
[the remark this answers is not in the document]
Comment (davemart-in) [reply extensions/blocks/recurring-payments/edit.jsx]: Thanks for that feedback! Reverted that change in https://github.com/Automattic/jetpack/commit/05e8e0a6b49fa460c6508f60ef55a508c91e54bd It's back to a single string now. 👍

**Review by jeherve — DISMISSED** [review]
Looking good now. Merge when ready!

File: extensions/blocks/recurring-payments/editor.scss [hunk 2]
```diff
@@ -1,18 +1,32 @@
 @import './view.scss';
 
-.wp-block-jetpack-membership-button {
-	font-family: -apple-system, BlinkMacSystemFont, Segoe UI, Roboto, Oxygen-Sans, Ubuntu, Cantarell,
-	Helvetica Neue, sans-serif;
+.wp-block-jetpack-recurring-payments {
+	font-family: -apple-system, BlinkMacSystemFont, Segoe UI, Roboto, Oxygen-Sans, Ubuntu, Cantarell, Helvetica Neue, sans-serif;
```
Comment (simison) [inline extensions/blocks/recurring-payments/editor.scss]: You can re-use the Gutenberg variable for this font-family:

```
@import '../../shared/styles/gutenberg-variables.scss';

font-family: $default-font;
```

Example here:

https://github.com/Automattic/jetpack/blob/8361c7010843b88776a67909db2369c3661bcdc2/extensions/blocks/simple-payments/editor.scss#L2-L5

Comment (artpi) [reply extensions/blocks/recurring-payments/editor.scss]: Done!

**Review by artpi — APPROVED** [review]
I implemented the imported font. Looks good now.

**Review by jeherve — APPROVED** [review]
Should be good to merge now!

**Pull request merged**